Qt6 build system prerequisites

Be-ing opened the pull request:
A bit of refactoring before adding an option to build with Qt6 (#4051)

Be-ing committed:
CMake: use ${QT_VERSION_MAJOR} instead of hardcoding Qt5

to prepare for Qt6 support

uklotzde commented on CMakeLists.txt:
```diff
@@ -2091,7 +2092,7 @@ elseif(WIN32)
       crypt32
 
-      # New libraries required by Qt5.
+      # New libraries required by Qt${QT_VERSION_MAJOR}.
```
Please keep the original "Qt5" string. Replacing it with the template is wrong.

Be-ing replied:
good catch

uklotzde replied:
Some of those comments don't provide any value. If they no longer apply please delete or rephrase them. Remainders from the Qt4/Qt5 migration.

uklotzde commented:
Any ideas why the macOS build is failing now?

Be-ing commented:
Not really, I still need to work on that

Be-ing commented:
Maybe this should wait until switching to vcpkg for macOS dependencies with Qt statically linked.

Be-ing commented on CMakeLists.txt:
```diff
@@ -2096,7 +2096,8 @@ target_link_libraries(mixxx-lib PRIVATE mixxx-proto)
 
 # Qt
-find_package(Qt5
+find_package(QT NAMES Qt5 COMPONENTS Core REQUIRED)
```
```suggestion
find_package(QT NAMES Qt6 COMPONENTS Core REQUIRED)
```
That is all that should be needed after this to test building with Qt6.

Be-ing committed:
CMake: remove legacy path for fixup_bundle

That was needed because the old buildserver scripts installed Qt
to a separate prefix from the rest of Mixxx's dependencies. vcpkg
does not do that.

Be-ing commented:
This is no longer failing on copying the Qt plugins into the macOS package now that we are linking Qt statically on macOS with vcpkg. Ready for review.

uklotzde commented on CMakeLists.txt:
```diff
@@ -2932,19 +2934,19 @@ if(APPLE AND MACOS_BUNDLE)
-        install_qt5_plugin(Qt5::QGifPlugin BUNDLE_LIBS "${MIXXX_INSTALL_PREFIX}")
-        install_qt5_plugin(Qt5::QICOPlugin BUNDLE_LIBS "${MIXXX_INSTALL_PREFIX}")
-        install_qt5_plugin(Qt5::QSQLiteDriverPlugin BUNDLE_LIBS "${MIXXX_INSTALL_PREFIX}")
+        install_qt5_plugin(QCocoaIntegrationPlugin BUNDLE_LIBS "${MIXXX_INSTALL_PREFIX}")
```
Should this macro be renamed to `install_qt_plugin` without any version number?

Be-ing replied:
Should we just remove it now that we're not using it?

daschuer replied:
The rest of the build system is written with the assumption that a dynamic build should continue to work. 
In this sense I would keep it, even though, we can't verify it and it will suffer rotting.
But it will definitely be helpfull in case we want to switch to a dynamic build later.

Be-ing replied:
I think it would be better to remove it. Yes, we need to support dynamic linking, but we don't need to do so for macOS.

Be-ing replied:
@uklotzde what do you think about removing this no longer used code?

uklotzde replied:
Ok. If it is not used then we should probably remove it. Less and simpler is better.

Be-ing committed:
CMake: remove unused code for copying Qt plugins to macOS bundle

This is not needed since switching to vcpkg and linking Qt
statically.

uklotzde reviewed (approved):
Nice, thank you! LGTM
